src/tide/include/math.ts

```diff
@@ -281,17 +281,17 @@ module _Math {
      * @param values the values to find the closest from
      */
     export function closest(x: number, values: number[]): number {
-        let minDistance = Infinity;
-        let index: number;
-        values.forEach((v, i) => {
-            const d = Math.abs(v - x);
-            if (d < minDistance) {
-                minDistance = d;
-                index = i;
-            }
-        });
+        let minDistance = Number.MAX_VALUE;
+        let index = 0;
+    
+        for (let i = 0; i < values.length; i++) {
+            const d = Math.abs(values[i] - x);
+            (d < minDistance) && (minDistance = d, index = i);
```

Comment:
It took me a while to understand what this line does. Does the execution speed gain really worth it?